src/python/pants/backend/python/lint/ruff/rules.py

```diff
@@ -31,20 +33,54 @@ class RuffFixRequest(FixTargetsRequest):
 
     @classproperty
     def tool_name(cls) -> str:
-        return "ruff --fix"
+        return "ruff check --fix"
+
+    @classproperty
+    def tool_id(self) -> str:
```

Comment:
I know that tool IDs are used in the [`determine_specified_tool_ids`](https://github.com/pantsbuild/pants/blob/main/src/python/pants/core/goals/multi_tool_goal_helper.py#L75) function, but does modifying the tool ID have any breaking implications elsewhere for end users?

Reply:
the docstring says it is used for tool selection
```python
    @classproperty
    def tool_id(cls) -> str:
        """The "id" of the tool, used in tool selection (Eg --only=<id>)."""
        return cls.tool_subsystem.options_scope
```
I'm pretty sure we need to use "ruff-check" and "ruff-format" tool_id-s for everything related to tool selection, so it must be right.

Reply:
Got it. So this will be a breaking change for folks who are using `ruff` as the tool ID - I don't know if that is necessarily a gating issue here but definitely something we would need to call out in release notes. Since 2.20 has not yet hit a release version it might be all right, but I'll let a maintainer chime in.

Reply:
Yes, but adding ruff formatter enabled by default is a breaking change in the first place. I started fixing it because it has broken my black+ruff setup

Reply:
This is a fair counter - in general, I'm not sure if this or #20343 is the better solution to this issue given that underneath, it is the same tool in all cases (and any tool we add in future that is both a fixer and a formatter will suffer from this issue as well).

Reply:
One way we could solve this it to ban duplicate tool_id in the core. We could add a check for a unique tool_id and fail if we find 2 identical tool_id-s

Reply:
> One way we could solve this it to ban duplicate tool_id in the core

I think the challenge with this is that we'd then have to make changes to `tool_id` for all tools in the Pants codebase, and not just Ruff?

That feels like a pretty heavy lift across the codebase compared to #20343, which allows users to benefit from this behavior change (fixers respecting `[fmt].only`) without needing to make massive changes elsewhere in their configuration.

Reply:
Makes sense. But the current pr is still needed to control `ruff check` lint and `ruff format --check` lint. Right now it's impossible to enable one of them and disable the other, and https://github.com/pantsbuild/pants/discussions/17237 doesn't help with that.